Compiler-Internals.md

```diff
@@ -39,8 +48,19 @@ It checks whether a symbol has an entry in the merged-symbols table each time be
 
 ### Type argument fixing
 
-# Emitter
+# Transformer
+
+The transformer is nearing completion to replace the emitter.
+The change in name is because the *emitter* translated TypeScript to JavaScript.
+The *transformer* transforms TypeScript or JavaScript (various versions) to JavaScript (various versions) using various module systems.
+The input and output are basically both trees from the same AST type, just using different features.
+There is still a small printer that writes any AST back to text.
 
 ## Rewriting & synthesized nodes
 
-## Sourcemap generation
\ No newline at end of file
+## Sourcemap generation
+
+# Language service
+
+The language service provides an API for clients to provide edit-time information about a program.
+Point it at a directory and it will maintain compilation information about the source in that directory.
```

Comment:
the language service? This is probably easier to get in the formatted version that merges all sentences into a paragraph.